docs/development.rst

```diff
@@ -26,157 +121,606 @@ Quickstart
 
-****************************
-Continuous integration tests
-****************************
+  $ pre-commit install
+
+See the :ref:`sec_development_workflow_git` section for detailed information
+on the recommended way to use git and GitHub.
+
+.. _sec_development_workflow:
 
-Three different continuous integration providers are used, which run different
-combinations of tests on different platforms:
+********
+Workflow
+********
 
-1. A `Github action <https://help.github.com/en/actions>`_ runs `pre-commit
-   <https://pre-commit.com/>`_ to run a variety of code style and quality checks.
+.. _sec_development_workflow_git:
 
-2. `Travis CI <https://travis-ci.org/>`_ runs tests on Linux and OSX using the
-   `Conda <https://conda.io/docs/>`__ infrastructure for the system level
-   requirements. All supported versions of Python are tested here.
+------------
+Git workflow
+------------
 
-3. `CircleCI <https://circleci.com/>`_ Runs all Python tests using the apt-get
-   infrastructure for system requirements. Additionally, the low-level tests
-   are run, coverage statistics calculated using `CodeCov <https://codecov.io/gh>`__,
-   and the documentation built.
+1. Make your own `fork <https://help.github.com/articles/fork-a-repo/>`_
+   of the ``tskit`` repository on GitHub, and
+   `clone <https://help.github.com/articles/cloning-a-repository/>`_
+   a local copy.
+2. Make sure that your local repository has been configured with an
+   `upstream remote <https://help.github.com/articles/configuring-a-remote-for-a-fork/>`_.
+3. Create a "topic branch" to work on. One reliable way to do it
+   is to follow this recipe::
 
-4. `AppVeyor <https://www.appveyor.com/>`_ Runs Python tests on 32 and 64 bit
-   Windows using conda.
+    $ git fetch upstream
+    $ git checkout upstream/master
+    $ git checkout -b topic_branch_name
 
-.. _sec_development_code_style:
+4. As you work on your topic branch you can add commits to it. Once you're
+   ready to share this, you can then open a `pull request
+   <https://help.github.com/articles/about-pull-requests/>`__. Your PR will
+   be reviewed by some of the maintainers, who may ask you to make changes.
+   If you'd like to open the PR but feel that the code isn't ready
+   for review yet, please use the "Draft" option on GitHub.
+5. If your topic branch has been around for a long time and has gotten
+   out of date with the main repository, we might ask you to
+   `rebase <https://help.github.com/articles/about-git-rebase/>`_
+   or to "squash" your commits into one.
+
+Please follow `this guide
+<https://stdpopsim.readthedocs.io/en/stable/development.html#rebasing>`_
+for step-by-step instructions on rebasing and squashing commits.
+
+.. _sec_development_workflow_pre_commit:
+
+-----------------
+Pre-commit checks
+-----------------
+
+On each commit a `pre-commit hook <https://pre-commit.com/>`_  will run
+that checks for violations of code style and other common problems.
```

Comment:
might be good to demystify this a bit, maybe inserting "(for instance, the pre-commit hook runs `black <https://github.com/psf/black>`_, which ensures the python code passes `PEP8 <https://www.python.org/dev/peps/pep-0008/>`_)"

Reply:
ah - I see this is explained more below; maybe say (see below)